controllers/managedocs_controller.go

```diff
@@ -1531,10 +1531,6 @@ func (r *ManagedOCSReconciler) reconcileEgressNetworkPolicy() error {
 				return fmt.Errorf("Unable to get DeadMan's Snitch secret: %v", err)
 			}
 		}
-		dmsURL := string(r.deadMansSnitchSecret.Data["SNITCH_URL"])
-		if dmsURL == "" {
-			return fmt.Errorf("DeadMan's Snitch secret does not contain a SNITCH_URL entry")
-		}
 		snitchURL, err := url.Parse(string(r.deadMansSnitchSecret.Data["SNITCH_URL"]))
 		if err != nil {
 			return fmt.Errorf("Unable to parse DMS url: %v", err)
```

Comment:
Why are we removing the specific check for an empty entry? I would prefer to have a more targeted error than a generic one. If the desire was to not parse twice then the solution should be:
```
snitchStr := string(r.deadMansSnitchSecret.Data["SNITCH_URL"])
if snitchStr == "" {
	return fmt.Errorf("DeadMan's Snitch secret does not contain a SNITCH_URL entry")
}
snitchURL, err := url.Parse(snitchStr)
if err != nil {
	return fmt.Errorf("Unable to parse DMS url: %v", err)
}
```